covid/transform.py

```diff
@@ -518,6 +518,28 @@ def transform_covidtracking_data(df):
     return df
 
 
+def transform_cdc_ili_data(df):
+    # Use the first row of the index as the index names.
+    df.index.names = df.index[0]
+
+    # Drop the row containing the labels, and set a new multi-index using the region, year, and week.
+    df = df.iloc[1:].reset_index()
+
+    # Validate that the only region type is states to sanity check data.
+    assert set(df["REGION TYPE"].unique()) == {"States"}
+
+    # Create a new column that combines the `YEAR` and `WEEK` column.
+    # Note: the `-1` sets these timestamp at Monday of each week.
+    df["_WEEK_TIMESTAMP"] = pd.to_datetime(
+        df["YEAR"] + "-W" + df["WEEK"] + "-1", format="%Y-W%W-%w"
+    )
+
+    # Create a new multi-index containing the state name (`REGION`) and timestamp of the week.
+    df = df.set_index(keys=["REGION", "_WEEK_TIMESTAMP"])
```

Comment:
Nice! Eventually we might want to change the name of the `state` field to `STATE_SOURCE_FIELD` so we can merge with the rest of the large dataframe. But we can also sort that out in the next PR.

Reply:
Ahh yes, that makes sense – I'll add that here actually.

Reply:
Also question – should I use the year and week (1 - 52) to create start of week timestamps or end of week timestamps? Currently I'm translating eg. `Week 1 of 2020` to the first day (Monday) of the week. Maybe it makes sense to convert that to the last day (Sunday) of the week?

Reply:
Ooh great question. Probably the last day of the week, yes -- that makes more sense with lining up the criteria.